### PR TITLE
(android) Fix camera keeps running after leaving scanner view (#442)

### PR DESCRIPTION
**Pull request by kotasjan**
Please test the changes. I am quite confident these changes will work, but I didn't have a chance to build and run it due to hurdles with the project build setup. 😕

### CODE REVIEW

**Comment by dpad85** [comment]
Thanks, I will look into that ASAP.

File: phoenix-android/src/main/kotlin/fr/acinq/phoenix/android/payments/ScanDataView.kt [hunk 1]
```diff
@@ -179,6 +180,9 @@ fun ReadDataView(
     Box(Modifier.fillMaxSize()) {
 
         if (initialInput.isNullOrBlank()) {
+            BackHandler {
+                scanView?.pause()
+            }
```
Comment (dpad85) [inline phoenix-android/src/main/kotlin/fr/acinq/phoenix/android/payments/ScanDataView.kt:185]: This changed can be rolled back if we use a `DisposableEffect`.

File: phoenix-android/src/main/kotlin/fr/acinq/phoenix/android/payments/ScanDataView.kt [hunk 2]
```diff
@@ -216,7 +220,10 @@ fun ReadDataView(
                 modifier = Modifier.fillMaxWidth(),
                 text = stringResource(id = R.string.btn_cancel),
                 icon = R.drawable.ic_arrow_back,
-                onClick = onBackClick
+                onClick = {
+                    scanView?.pause()
+                    onBackClick()
+                }
```
Comment (dpad85) [inline phoenix-android/src/main/kotlin/fr/acinq/phoenix/android/payments/ScanDataView.kt:226]: This change can be rolled back if we use a `DisposableEffect`.

**Review by dpad85 — CHANGES_REQUESTED** [review]
I was able to reproduce issue #442 on a Pixel device with Android 14 (though on a Pixel device with Android 12, the camera does not stay active all the time).

I also tested your changes, and pausing the scanner does fix the issue - however it needs to be applied on other cases (e.g. when copy-pasting). A better solution is to simply use a `DisposableEffect` in the `CameraPermissionsView.onPermissionGranted` callback, instead of a `LaunchedEffect`:

```kotlin
DisposableEffect(key1 = model) {
    if (model is Scan.Model.Ready && initialInput.isNullOrBlank()) scanView?.resume()
    onDispose {
        scanView?.pause()
    }
}
```

If you give me push authorisation on your branch, I can update your PR with that change.

**Comment by kotasjan** [comment]
> I was able to reproduce issue #442 on a Pixel device with Android 14 (though on a Pixel device with Android 12, the camera does not stay active all the time).
> 
> I also tested your changes, and pausing the scanner does fix the issue - however it needs to be applied on other cases (e.g. when copy-pasting). A better solution is to simply use a `DisposableEffect` in the `CameraPermissionsView.onPermissionGranted` callback, instead of a `LaunchedEffect`:
> 
> ```kotlin
> DisposableEffect(key1 = model) {
>     if (model is Scan.Model.Ready && initialInput.isNullOrBlank()) scanView?.resume()
>     onDispose {
>         scanView?.pause()
>     }
> }
> ```
> 
> If you give me push authorisation on your branch, I can update your PR with that change.

Thank you for the feedback. You are tight with `DisposableEffect` - it is a much cleaner solution. 👍 I have added you to contributors, so you should be able to push your changes now.

**Review by dpad85 — APPROVED** [review]
Thanks again @kotasjan

**Pull request merged**